epochX/cudacpp/CODEGEN/PLUGIN/CUDACPP_SA_OUTPUT/model_handling.py

```diff
@@ -888,12 +926,30 @@ def super_generate_parameters_class_files(self):
             replace_dict['dcoupsetdcoup2'] = '      // (none)'
             replace_dict['dcoupoutdcoup2'] = ''
         # Require HRDCOD=1 in EFT and special handling in EFT for fptype=float using SIMD
+        replace_dict['bsmdefine'] = '#define MGONGPUCPP_NBSMINDEPPARAM_GT_0 1' if len( bsmparam_indep_real_used ) > 0 else '#undef MGONGPUCPP_NBSMINDEPPARAM_GT_0'
         if self.model_name[:2] == 'sm' :
```

Comment:
This condition is really dangerous since some model can be 
sm_with_zprime

Reply:
Ah... interesting. What do you suggest instead? Spell out the full SM model name and compare to that string? On the other hand sm_nobmass requires no special handling.

Maybe I should check what happens if I do this "extra BSM handling" also for all SM processes. I guess that it will add some unused and unnecessary boilerplate code, but will not change the functionality.

Suggestions?